### DISCUSSION TITLE
Getting Invalid workflow file error GitHub Action

### DISCUSSION THREAD

**Discussion by kuddusrhce** [community/community#45756] [Actions]
### Select Topic Area

Product Feedback

### Body

Hello Team,

We are using GitHub Action with a self-hosted runner and we are getting the below error during CI.

Error: The workflow is not valid. .github/workflows/my.yaml (Line: xx, Col: xx): Exceeded max expression length.

Can you please let us know how to fix it?

**Comment by jge162** [comment]
@kuddusrhce Your workflow file exceeds the maximum limit for YAML expressions in GitHub Actions. 

consider making these changes ->

- Break the long workflow into `smaller workflows`, and trigger them from a `main workflow.`
- Use [YAML anchors](https://support.atlassian.com/bitbucket-cloud/docs/yaml-anchors/) and references to reduce duplication.
- Reduce the number of actions in the workflow by optimizing the code or combining similar steps as aforementioned. 

Anything else please do not hesitate to asks.

**Comment by pratiyush05** [comment]
Hi @jge162 
I am getting the same error while running in self-hosted runner - Exceeded max expression length 21000 while pointing to a "run: | " expression in a workflow . Could you please tell whether this limit is for a single "run expression" or for the whole workflow . 
It is quiet complicated for me to divide that workflow into multiple workflows .For dividing ,I would need to trigger workflows from shell script of main workflow .
